cmd/controller/controller.go

```diff
@@ -222,7 +222,7 @@ func (c *CmdOpts) startController(ctx context.Context) error {
 			adminClientFactory))
 	}
 
-	if c.NodeConfig.Spec.API.ExternalAddress != "" && !c.NodeConfig.Spec.API.TunneledNetworkingMode {
+	if !c.SingleNode && !c.NodeConfig.Spec.API.TunneledNetworkingMode {
```

Comment:
hmm, I think this on was correct before. The k0s custom `EndpointReconciler` creates the `Endpoints` for kube api svc based on `externalAddress` via DNS resolution. Thus I don't think we should enable it unless there actually is `externalAddress`. Or am I missing something?

Reply:
Hm, I am trying to figure out what is the correct use case when we want to be:
- HA
- no external address
- tunneled network

at the same time.

I am even thinking, may be the logic should be totally different: use this reconciler if TunneledNetworkingMode, and add some validation somewhere, that if TunneledNetworkingMode is true but ExternalAddress is empty - fail the bootstrap. Wdyt?

Reply:
There's many variables in this game for sure. :)

>  if TunneledNetworkingMode is true but ExternalAddress is empty - fail the bootstrap

That sounds about right and good thing to do in general, fails the bootstrap for config combos that are not gonna work properly. (dunno if it starts to creep a bit out of this PRs scope though)

Reply:
I think it might be better to get it out of that PR, so I just reverted the line you have commneted